nncf/quantization/algorithms/weight_compression/awq.py

```diff
@@ -132,156 +132,40 @@ def apply(
         :return: A resulting model.
         """
         self._set_backend_entity(model, wc_backend_entity)
-        matches = []
 
-        inference_nncf_graph = transform_to_inference_graph(deepcopy(graph), [], [], [], [])
-        nx_graph = inference_nncf_graph.get_nx_graph_copy()
-        for _, pattern_graph in self._patterns.items():
-            matches.extend(find_subgraphs_matching_pattern(nx_graph, pattern_graph(), strict=False))
-
-        if len(matches) == 0:
-            nncf_logger.info("No matching patterns were found for applying AWQ algorithm, it will be skipped.")
+        awq_data = self._get_awq_data(graph, all_weight_params, nodes_to_compress)
+        if len(awq_data) == 0:
             return model
 
         transformation_layout = TransformationLayout()
         model_transformer = ModelTransformerFactory.create(model, inplace=True)
 
-        awq_data = {}
-        name_mapping = {wp.weight_name: idx for idx, wp in enumerate(all_weight_params)}
-
-        for match in matches:
-            nncf_node = graph.get_node_by_key(match[-1])
-            if not self._backend_entity.is_node_with_weights(nncf_node, graph):
-                continue
-
-            target_node_names = []
-            for weight_op_friendly_name, _ in self._backend_entity.get_weight_names_and_port_ids(nncf_node, graph):
-                target_node_names.append(weight_op_friendly_name)
-
-            # skip node if it is in IgnoredScope or should not be compressed
-            if target_node_names[-1] not in name_mapping:
-                continue
-
-            weight_params = all_weight_params[name_mapping[target_node_names[-1]]]
+        is_data_free = statistics is None
```

Comment:
The description and type hint for `statistics` parameter should probably also be updated.